### PR TITLE
add typescript support to e2e tests

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
Add Typescript support to e2e tests

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45386

#### Related Pull Requests
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/tests/support/api/setup.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {UserProfile} from '@mattermost/types/users';
```
Comment (neallred) [inline e2e/cypress/tests/support/api/setup.ts:4]: One pain point of reviewing is that ofen, for the support files, it isn't recognized that the old `.js` & `d.ts` files are a rename to `.ts`, so it makes the diffs larger than they really are.

File: e2e/cypress/tests/integration/account_settings/profile/profile_picture_spec.ts [hunk 2]
```diff
@@ -10,6 +10,8 @@
 // Stage: @prod
 // Group: @account_setting
 
+import "cypress-file-upload";
```
Comment (neallred) [inline e2e/cypress/tests/integration/account_settings/profile/profile_picture_spec.ts:13]: This is a temporary add until `tests/support/index.ts` is converted to Typescript: it already has this import, which exends cypress with `attachFiles`. Until then, the easiest way is to redeclare the import.

**Review by JulienTant — CHANGES_REQUESTED** [review]
LGTM,  just found this little quark but I'll turn my review green upon fix!

**Review by JulienTant — APPROVED** [review]

**Comment by saturninoabril** [comment]
@mickmister I've added you to further check on how this will play with E2E tests of https://github.com/mattermost/mattermost-plugin-apps which imports this repo as its dependency with custom commands/helpers.

**Review by saturninoabril — APPROVED** [review]
Thanks @neallred! Looks good to me.

**Pull request merged**